packages/gatsby-theme-learning/src/components/profile-complete-guard.tsx

```diff
@@ -0,0 +1,55 @@
+import { useAuth0 } from '@auth0/auth0-react';
+import type { User } from '@auth0/auth0-react';
+import { useContext, useEffect } from 'react';
+import { LearningContext } from './learning-context';
+import { useAuthToken } from '../hooks/use-auth-token';
+import ConfigContext, {
+  EFeatureFlag,
+  isFeatureEnabled,
+} from './config-context';
+
+export type TProfileFormValues = {
+  firstName: string;
+  lastName: string;
+  company: string;
+};
+
+const ProfileModalGuard = () => {
+  const { isAuthenticated, user } = useAuth0();
+  const { getAuthToken } = useAuthToken();
+  const { auth0Domain, features } = useContext(ConfigContext);
+  const { updateProfile } = useContext(LearningContext);
+
+  const isModalFeatureEnabeld = isFeatureEnabled(
+    EFeatureFlag.CompleteProfileModal,
+    features
+  );
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const authToken = await getAuthToken();
+        const getUserApiEndpoint = `https://${auth0Domain}/api/v2/users/${user?.sub}`;
+        const data = await fetch(getUserApiEndpoint, {
+          method: 'GET',
+          headers: {
+            Accept: 'application/json',
+            Authorization: `Bearer ${authToken}`,
+          },
+        });
+        if (data.status !== 200) {
+          throw new Error(`Error fetching user profile for user ${user?.sub}`);
+        }
+        const userData = (await data.json()) as User;
+        updateProfile(userData);
+      } catch (error) {
+        console.error(error);
+      }
+    }
+    if (isModalFeatureEnabeld && isAuthenticated) {
```

Comment:
I'm not sure if I fully understood the code path here. This looks like it would call the API every time the app is loaded, even if firstname, lastname and company are already properly set as made available by the Auth0  login.   
@FFawzy  maybe I'm misled here and the Auth0 session doesn't actually provide these values. 

Maybe I'm sticking my finger into a more general thing here - the user profile with first name, last name etc is application state whose master information is on the server side and it's being shown in many places in the UI (e.g. avatar, the "hey welcome back Gabriele" thing we have on the board etc).   In other places we used SWR to centrally cache it and deduplicate the requests.   Here it may be a bit more fiddly as the avatar is actually pulling the initial info from the Auth0 token's custom claims - that's btw also maybe why the avatar does not update if I change my name in the form.  Could it make sense to use the Auth0 provided data as some kind of initialization of the local cache (like SWR) but from then on have all components rendering it use that cache only? 

Sounds like something we better discuss together (or I really missed something)

Reply:
> Sounds like something we better discuss together (or I really missed something)

Let's discuss this together. Some inputs:
- At login, the `user` object that auth0 returns is missing the `company` field, that's why I need to always fetch the user profile to have a complete view of what fields might be missing
- Avatar component is getting most of the information displayed from custom claims such as `https://docs.commercetools.com/display_name` and `https://docs.commercetools.com/global_account_name` it's not directly fetching data from the auth0 user profile. 
- I consider the user profile as data with very low update rate: the user will typically update those information only when logs in the first time and probably never again, or something close to it. For this reason I believe that storing this data in context once at login is a better approach rather than using SWR.

Reply:
> 
@FFawzy  it shuold be trivial to add the company here: https://github.com/commercetools/learning-tech/blob/main/auth0/add-user-data-to-token-action.js#L7

Reply:
- we can definitely add the missing `company` field to the token.
- @nkuehn I'll add a task for me to document these somewhere.
- the million dollar question here is, does the `user` object from the `useAuth0` hook ... update by itself after updating a user through the api ? or is there a specific parameter that triggers this? @gabriele-ct did you try this ?
if yes, then it will be easier to leave out your fetch

Reply:
I've implemented the following changes:
- the user profile is read from the token upon login and stored in the context.
- All the parts of the application where profile informations are needed are now using LearningContext (apart from avatar which I'll update once Timon's changes are merged)
- Upon user profile update, the updated profile returned from the API is used to update the app LearningContext